FloripaPublicTransportation/MapViewController.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  MapViewController.swift
+//  FloripaPublicTransportation
+//
+//  Created by Marcelo Gobetti on 2/22/16.
+//  Copyright © 2016 Marcelo Gobetti. All rights reserved.
+//
+
+import MapKit
+
+class MapViewController: UIViewController, MKMapViewDelegate, CLLocationManagerDelegate {
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet private weak var mapView: MKMapView!
+    @IBAction private func addPin(sender: UILongPressGestureRecognizer) {
+        guard sender.state == .Began else {
+            return
+        }
+        
+        self.mapView.removeAnnotations(self.mapView.annotations)
+        let touchCoordinates = self.mapView.convertPoint(sender.locationInView(self.mapView), toCoordinateFromView: self.mapView)
+        let annotation = MKPointAnnotation()
+        annotation.coordinate = touchCoordinates
+        
+        let geocoder = CLGeocoder()
+        geocoder.reverseGeocodeLocation(CLLocation(latitude: annotation.coordinate.latitude, longitude: annotation.coordinate.longitude), completionHandler: {
```

Comment:
Actually, I haven't noticed that it takes a noticeable time to process that part, but I believe you are right: now I also see that the self.mapView.addAnnotation call should have been made inside the completion handler as well, and then maybe I would be able to get rid of the ugly timer I have used for automatically popping the annotation's callout. What do you suggest, an UIActivityIndicator over the map?